vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -37,39 +38,17 @@ class Graph(VizroBaseModel):
     """
 
     type: Literal["graph"] = "graph"
-    figure: CapturedCallable = Field(..., import_path=px)
+    figure: CapturedCallable = Field(
+        ..., import_path=px
+    )  # TODO: needs to be refactored so plotly-independent or extendable - No, as within the boundaries of Graph model
     actions: List[Action] = []
 
+    # Component properties for actions and interactions
+    _output_property: str = PrivateAttr("figure")
+
     # Re-used validators
     _set_actions = _action_validator_factory("clickData")
-
-    @validator("figure")
-    def process_figure_data_frame(cls, figure, values):
-        data_frame = figure["data_frame"]
-
-        # Enable running px.scatter("iris") from the Python API and specification of "data_frame": "iris" through JSON.
-        # In these cases, data already exists in the data manager and just needs to be linked to the component.
-        if isinstance(data_frame, str):
-            data_manager._add_component(values["id"], data_frame)
-            return figure
-
-        # Standard case for px.scatter(df: pd.DataFrame).
-        # Extract dataframe from the captured function and put it into the data manager.
-        dataset_name = str(id(data_frame))
-
-        logger.debug("Adding data to data manager for Graph with id %s", values["id"])
-        # If the dataset already exists in the data manager then it's not a problem, it just means that we don't need
-        # to duplicate it. Just log the exception for debugging purposes.
-        try:
-            data_manager[dataset_name] = data_frame
-        except ValueError as exc:
-            logger.debug(exc)
-
-        data_manager._add_component(values["id"], dataset_name)
-
-        # No need to keep the data in the captured function any more so remove it to save memory.
-        del figure["data_frame"]
-        return figure
+    _validate_callable = validator("figure", allow_reuse=True, always=True)(_process_callable_data_frame)
```

Comment:
```suggestion
    _validate_callable = validator("figure", allow_reuse=True)(_process_callable_data_frame)
```

This shouldn't be needed since `figure` is a mandatory field.